JavaScript part 3 - Assignments.js

```diff
@@ -0,0 +1,274 @@
+    return end - start;
+}
+
+let timePockets = benchmark(pockets, 'glasses');
+let timeHead = benchmark(head, 'glasses');
+
+console.log("Time taken for pockets.glasses:", timePockets, "milliseconds");
+console.log("Time taken for head.glasses:", timeHead, "milliseconds");
+
+
+if (timePockets < timeHead) {
+    console.log("pockets.glasses is faster");
+} else if (timeHead < timePockets) {
+    console.log("head.glasses is faster");
+} else {
+    console.log("Both property lookups have similar speed");
+}
+
+/*
+We have rabbit inheriting from animal.
+If we call rabbit.eat(), which object receives the full property: animal or rabbit?
+*/
+
+let animal = {
+    eat() {  
+      this.full = true;    
+    }  
+  };
+  
+  let rabbit = {  
+    __proto__: animal  
+  };
+
+console.log("Before call:", rabbit);
+rabbit.eat();
+console.log("After call:", rabbit);
+
+//rabbit is called
+//js looks for eat method but cant find it so it follows the prototype chain 
+//finds eat method in animal object
+//method is executed while referring to the rabbit and full property is added to rabbit object
+
+/*
+Task 1. The Clock class (see the sandbox) is written in functional style. Rewrite it in the “class” syntax.
+function Clock({ template }) {
+
+    let timer;
+    function render() {
+     let date = new Date();
+      let hours = date.getHours();
+     if (hours < 10) hours = '0' + hours;
+      let mins = date.getMinutes();
+     if (mins < 10) mins = '0' + mins;
+      let secs = date.getSeconds();
+     if (secs < 10) secs = '0' + secs;
+      let output = template
+       .replace('h', hours)
+       .replace('m', mins)
+       .replace('s', secs);
+      console.log(output);
+   }
+   
+    this.stop = function() {
+     clearInterval(timer);
+   };
+
+    this.start = function() {
+     render();
+     timer = setInterval(render, 1000);
+   };
+
+  }
+  let clock = new Clock({template: 'h:m:s'});
+ clock.start();
+*/
+
+class Clock {
+    constructor({ template }) {
+      this.template = template;
+      this.timer = null;
+    }
+  
+    render() {
+      let date = new Date();
+      let hours = date.getHours();
+      if (hours < 10) hours = '0' + hours;
+      let mins = date.getMinutes();
+      if (mins < 10) mins = '0' + mins;
+      let secs = date.getSeconds();
+      if (secs < 10) secs = '0' + secs;
+      let output = this.template
+        .replace('h', hours)
+        .replace('m', mins)
+        .replace('s', secs);
+      console.log(output);
+    }
+  
+    stop() {
+      clearInterval(this.timer);
+    }
+  
+    start() {
+      this.render();
+      this.timer = setInterval(() => this.render(), 1000);
+    }
+  }
+  
+  let clock = new Clock({ template: 'h:m:s' });
+  clock.start();
+  
+/*
+Write a function count(obj) that returns the number of properties in the object:
+*/
+
+let user = {
+    name: 'John',
+    age: 30
+  };
+
+function count(obj){
+   return this.obj = Object.keys(obj).length
+
```

Comment:
In your function, return this.obj = Object.keys(obj).length attempts to assign the length of Object.keys(obj) to this.obj. This is unnecessary and incorrect for calculating the count of properties in the object.

Just use the Global Object and it's method: 

function count(obj) {
  return Object.keys(obj).length;
}